source/slang/slang-type-layout.cpp

```diff
@@ -3899,6 +3899,19 @@ void TypeLayout::removeResourceUsage(LayoutResourceKind kind)
     }
 }
 
+void VarLayout::removeResourceUsage(LayoutResourceKind kind)
+{
+    Int infoCount = resourceInfos.getCount();
+    for( Int ii = 0; ii < infoCount; ++ii )
+    {
+        if( resourceInfos[ii].kind == kind )
+        {
+            resourceInfos.removeAt(ii);
```

Comment:
Presumably you could use fastRemoveAt - doesn't seem to be an ordering of resourceInfos.

Reply:
I don't think we have any code that relies on the ordering of the resourceInfos, but I've tried not to write code that would disturb their order once established. There are two different future directions where might want to guarantee an ordering of the resourceInfos, and I want to leave the door open for them:

1. We might decide that the first resourceInfo should always be the "primary" resource usage for a type. For example we might want a `ConstantBuffer<X>` to always report the constant buffer register/binding as the first resourceInfo (as a guarantee to the reflection API user), whether or not `X` contains textures or other resources.

2. Probably more interesting: if we guarantee that the resourceInfos are always sorted by `LayoutResourceKind` then we can look up a particular resource kind with a binary search instead of the linear search we do today.